packages/core/src/clients/ITopicClient.ts

```diff
@@ -24,7 +23,17 @@ export interface ITopicClient {
   ): Promise<TopicSubscribe.Response>;
 
   listWebhooks(cache: string): Promise<ListWebhooks.Response>;
-  putWebhook(webhook: Webhook): Promise<PutWebhook.Response>;
-  deleteWebhook(id: WebhookId): Promise<DeleteWebhook.Response>;
-  getWebhookSecret(id: WebhookId): Promise<GetWebhookSecret.Response>;
+  putWebhook(
+    cacheName: string,
+    webhookName: string,
+    options: PutWebhookCallOptions
```

Comment:
I'm saying that all the other functions have `cacheName` and `topicName` as their first two arguments.

Those are the two arguments I was advocating for consistency on.

If you want to put the `webhookName` and friends into `Options` on any of these PRs I'm okay with it.

Reply:
for the other methods which have `cacheName` and `topicName`, they are for `topics`, so `topicName` is a first class citizen. For `webhooks`, `webhookName` is a first class citizen, so I think the pattern is consistent

Reply:
i disagree. this re-ordering of the arguments is the thing that would be the most likely to cause users to accidentally put the wrong string values in the wrong order.

there is a hierarchy to these things:

* Cache is the top level
* Topics are next - you can't have a topic without a cache
* Webhooks are next - you can't have a webhook without a topic

the arguments should appear in that order. whether you want to make the webhookName argument a string or part of an `Options` is up to you, but the order should be consistent and reflect the hierarchy.

Reply:
`DeleteWebhook` and `getWebhookSecret` both have `cache: String, webhookName: string` since that is what we use to identify the webhook. If we have `putWebhook` be `cache: string, topic: string, options: Options` then it will be inconsistent with `deleteWebhook` and `getWebhookSecret`

Reply:
if you don't agree with what i'm saying here, feel free to start a notion doc or slack thread with the proposed 5 signatures in it and we can solicit some additional opinions.